Source/WebCore/platform/network/NetworkStorageSession.cpp

```diff
@@ -269,7 +271,7 @@ bool NetworkStorageSession::hasStorageAccess(const RegistrableDomain& resourceDo
         }
 
         auto it = m_pairsGrantedCrossPageStorageAccess.find(firstPartyDomain);
-        if (it != m_pairsGrantedCrossPageStorageAccess.end() && it->value.contains(resourceDomain))
+        if (it != m_pairsGrantedCrossPageStorageAccess.end() && it->value.isValidValue(resourceDomain) && it->value.contains(resourceDomain))
```

Comment:
Why do we need to validate resourceDomain? When is it invalid?

Reply:
I suppose it might be safe by the time we reach this point.